app/src/main/java/com/keylesspalace/tusky/fragment/NotificationsFragment.java

```diff
@@ -491,6 +498,84 @@ public void onLoadMore(int position) {
+     * status content is interacted with.
+     *
+     * @param isCollapsed Whether the status content is shown in a collapsed state or fully.
+     * @param position    The position of the status in the list.
+     */
+    @Override
+    public void onContentCollapsedChange(boolean isCollapsed, int position) {
+        if(position < 0 || position >= notifications.size()) {
+            Log.e(TAG, String.format("Tried to access out of bounds status position: %d of %d", position, notifications.size() - 1));
+            return;
+        }
+
+        NotificationViewData notification = notifications.getPairedItem(position);
+        if(!(notification instanceof NotificationViewData.Concrete)) {
+            if(notification == null) {
+                Log.e(TAG, String.format(
+                        "Tried to access notification but got null at position: %d of %d",
+                        position,
+                        notifications.size() - 1)
+                );
+            } else {
+                Log.e(TAG, String.format(
+                        "Expected NotificationViewData.Concrete, got %s instead at position: %d of %d",
+                        notification.getClass().getSimpleName(),
+                        position,
+                        notifications.size() - 1
+                ));
+            }
+
+            return;
+        }
+
+        StatusViewData.Concrete status = ((NotificationViewData.Concrete) notification).getStatusViewData();
+        if(status == null) {
+            Log.e(TAG, String.format(
+                    "Tried to access status in notification but got null at position: %d of %d",
+                    position,
+                    notifications.size() - 1)
+            );
+            return;
+        }
+
+        StatusViewData.Concrete updatedStatus = new StatusViewData.Builder(status)
+                .setCollapsed(isCollapsed)
+                .createStatusViewData();
+
+        NotificationViewData.Concrete concreteNotification = (NotificationViewData.Concrete) notification;
+        NotificationViewData updatedNotification = new NotificationViewData.Concrete(
+                concreteNotification.getType(),
+                concreteNotification.getId(),
+                concreteNotification.getAccount(),
+                updatedStatus,
+                concreteNotification.isExpanded()
+        );
+        notifications.setPairedItem(position, updatedNotification);
+        adapter.updateItemWithNotify(position, updatedNotification, false);
+
+        // Since we cannot notify to the RecyclerView right away because it may be scrolling
+        // we run this when the RecyclerView is done doing measurements and other calculations.
+        // To test this is not bs: try getting a notification while scrolling, without wrapping
+        // notifyItemChanged in a .post() call. App will crash.
+        recyclerView.post(() -> adapter.notifyItemChanged(position, notification));
```

Comment:
just wrap whole `updateItemWithNotify` in `post`

Reply:
I just wanted to avoid adding an entire level of nesting, since it's needed only for one method.